### PR TITLE
fix(aws-lambda-event-sources): Validate that SQS Queue visibility timeout is larger than Lambda timeout

### PR DESCRIPTION
**Pull request by Gtofig**
This PR adds compile time check when SQS Queue is added as event source to Lambda to validate that AWS Lambda timeout is smaller that Queue visibility timeout. AWS CloudFormation deployment will fail if Lambda timeout is larger than Queue visibility timeout when adding Queue as an event source.

To enable this change, PR also makes two related changes:
- In (aws-lambda) module - expose `timeout` prop in `IFunction` interface and all implementations
- In (aws-sqs) module - expose `visibilityTimeout` prop in `IQueue` interface and all implementation

closes https://github.com/aws/aws-cdk/issues/20527

----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Gtofig** [comment]
It's not clear to me if this is a (feat) or a (fix), but using (feat) results in a failed linter due to lack of README changes. I'm not sure this validation requires any README changes though. Please let me know how you want to proceed.

File: packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts
Comment (Gtofig) [reply packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts:71]: Yes, this is already failing at deploy time

Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts:71]: Great, then fix is accurate for this.

**Review by TheRealAmazonKendra — DISMISSED** [review]
Looks like this is going to need to update the integration tests to get this build passing. Please not that we cannot provide a meaningful review until the build is passing and our various checks are all succeeding. If you have any questions about these build failures, please let me know.

**Comment by Gtofig** [comment]
Thanks. I will try to clean this up and resubmit later this week

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
When you want to merge from main, please use `@Mergifyio update` because it won't dismiss my review/request for changes. That helps us know when we need to do another review or when we're still awaiting changes.

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Comment by Gtofig** [comment]
@TheRealAmazonKendra - I'm facing some difficulties. When I've tried running `yarn integ --update-on-failed` on `aws-lambda` package, but getting the following error in CloudFormation: `Replacement type updates not supported on stack with disable-rollback.`. 

Correct me if I'm wrong, but this looks unrelated to the changes I've made. Furthermore, I'm quite surprized that integration tests would need to be changed for my change. Can you help me understand how my change might have affected snapshots?

**Comment by TheRealAmazonKendra** [comment]
> @TheRealAmazonKendra - I'm facing some difficulties. I've tried running `yarn integ --update-on-failed` on `aws-lambda` package, but getting the following error in CloudFormation: `Replacement type updates not supported on stack with disable-rollback.`.
> 
> Correct me if I'm wrong, but this looks unrelated to the changes I've made. Furthermore, I'm quite surprized that integration tests would need to be changed for my change. Can you help me understand how my change might have affected snapshots?

So, what's going on with the failing tests are that this construct is being used in another test package. That test, however, isn't setting the values in a way that meets the validation you've created here. If you can't update the tests with running that command, you can also manually edit the test inputs and expected output. Because this failure is showing up in `ecs-service-extensions`, you'll want to find the tests that set both timeouts to 30 and adjust the input values. Then run `yarn integ --update-on-failed` in that package and it should fix the output for you. It's possible that this construct is being used elsewhere as well so do the same for each package where the error gets thrown. If you get more permission issues, let me know and I can help update the tests.

**Pull request closed without merging**
